Sources/MessagingInApp/Gist/Managers/MessageQueueManager.swift

```diff
@@ -87,7 +87,16 @@ class MessageQueueManagerImpl: MessageQueueManager {
     }
 
     func getInlineMessages(forElementId elementId: String) -> [Message] {
-        localMessageStore.filter { $0.value.elementId == elementId }.map(\.value).sortByMessagePriority()
+        let messages = localMessageStore.filter {
+            if $0.value.elementId == elementId {
+                if $0.value.doesHavePageRule() {
+                    return $0.value.doesPageRuleMatch(route: Gist.shared.getCurrentRoute())
+                }
+                return true
+            }
+            return false
+        }.map(\.value).sortByMessagePriority()
+        return messages
     }
```

Comment:
There are two ways inline messages are checked, fetched and shown to the user if found in the local storage: 
- Through polling ([EventBus event posted](https://github.com/customerio/customerio-ios/blob/edcd7765e3388ba4d16424a2c798149d09b4b15c/Sources/MessagingInApp/Gist/Managers/MessageQueueManager.swift#L143))
- When an [element id is set on a view](https://github.com/customerio/customerio-ios/blob/edcd7765e3388ba4d16424a2c798149d09b4b15c/Sources/MessagingInApp/Views/InAppMessageView.swift#L92) 

Either ways, the messages are fetched using  `getInlineMessages` method and then shown.